client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
-jest.mock( 'calypso/blocks/dismissible-card', () => {
-	return function DismissibleCard() {
-		return null;
-	};
-} );
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import preferencesReducer from 'calypso/state/preferences/reducer';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
+import { Banner } from '../index';
 
-jest.mock( 'calypso/lib/analytics/track-component-view', () => {
-	return function TrackComponentView() {
-		return null;
+jest.mock( 'calypso/state/analytics/actions', () => {
+	return {
+		recordTracksEvent: jest.fn( () => ( {
+			type: 'ANALYTICS_EVENT_RECORD',
+		} ) ),
 	};
 } );
 
-import { Card, Button } from '@automattic/components';
-import { shallow } from 'enzyme';
-import PlanPrice from 'calypso/my-sites/plan-price/';
-import { Banner } from '../index';
+function renderWithRedux( ui ) {
+	return renderWithProvider( ui, {
+		reducers: {
+			preferences: preferencesReducer,
+		},
+		initialState: { preferences: { remoteValues: {} } },
+	} );
+}
 
 const props = {
 	callToAction: null,
 	title: 'banner title',
 };
 
 describe( 'Banner basic tests', () => {
-	test( 'should not blow up and have proper CSS class', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( '.banner' ) ).toHaveLength( 1 );
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	test( 'should output Banner component with proper CSS class', () => {
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.firstChild ).toHaveClass( 'banner' );
```

Comment:
I wanted to add an `aria-label` and `role` to the parent `<div>` for the Banner component and  `queryByRole` or `getByLabelText`.  However, the parent is the Card component, which is used for many different purposes, so it doesn't seem like there would be a descriptive enough role or label to fit. 🤔 The Banner component also has many different uses (upselling, providing information, containing a link or button, etc.) so it also seems difficult to give a generic role/description. Any thoughts here?

Reply:
Yeah, I think the solution isn't always to add a `role` and or `aria-label`. I agree with your solution here to not add a `role` - there's likely not a suitable one that fits all `Banner` cases, as you stated above. While we could add a `banner` role, that's generally reserved for the header that repeats throughout the site, and could make more sense to be our masterbar instead. We could also add a `complementary` role, but if we head there we could just make the wrapper be an `aside`, and that would make it `complementary` implicitly. I'd personally keep it as-is, and would go with `container.firstChild` when looking the element up.